kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/codec/ByteBufAccessorImpl.java

```diff
@@ -138,13 +133,9 @@ public double readDouble() {
 
     @Override
     public byte[] readArray(int size) {
-        int remaining = buf.readableBytes();
-        if (size > remaining) {
-            throw illegalReadException(size, remaining);
-        }
-        byte[] dst = new byte[size];
-        buf.readBytes(dst, 0, size);
-        return dst;
+        byte[] array = ByteBufUtil.getBytes(buf, buf.readerIndex(), size, false);
```

Comment:
Doesn't this change semantics? It is safe.  Previously the code returned a copy of the bytes, whereas now (copy=false) it is potential returning the live bytes.    There's nothing in the contact io.kroxylicious.proxy.frame.ByteBufAccessor#readArray or org.apache.kafka.common.protocol.Readable#readArray that actually states what behaviour should be.   I notice that the ResponseData objects propagate the reference.  We should be cautious before making a change.

AFAIK, I think the change might be safe and I don't think it actually changes anything for the Filter Author.  If we decide to go this away, we should javadoc ByteBufAccessor accordingly.